SaitamaRobot/modules/approve.py

```diff
@@ -0,0 +1,179 @@
+
+@user_admin
+@run_async
+def approved(update, context):
+    message = update.effective_message
+    chat_title = message.chat.title
+    chat = update.effective_chat
+    msg = "The following users are approved.\n"
+    approved_users = sql.list_approved(message.chat_id)
+    for i in approved_users:
+        member = chat.get_member(int(i.user_id))
+        msg += f"- `{i.user_id}`: {member.user['first_name']}\n"
+    if msg.endswith("approved.\n"):
+        message.reply_text(f"No users are approved in {chat_title}.")
+        return
+    else:
+        message.reply_text(msg, parse_mode=ParseMode.MARKDOWN)
+
+
+@user_admin
+@run_async
+def approval(update, context):
+    message = update.effective_message
+    chat = update.effective_chat
+    args = context.args
+    user_id = extract_user(message, args)
+    member = chat.get_member(int(user_id))
+    if not user_id:
+        message.reply_text(
+            "I don't know who you're talking about, you're going to need to specify a user!")
+        return ""
+    if sql.is_approved(message.chat_id, user_id):
+        message.reply_text(
+            f"{member.user['first_name']} is an approved user. Locks, antiflood, and blocklists won't apply to them.")
+    else:
+        message.reply_text(
+            f"{member.user['first_name']} is not an approved user. They are affected by normal commands.")
+
+
+@run_async
+def unapproveall(update: Update, context: CallbackContext):
+    chat = update.effective_chat
+    user = update.effective_user
+    member = chat.get_member(user.id)
+    if member.status != "creator" and user.id not in DRAGONS:
+        update.effective_message.reply_text(
+            "Only the chat owner can unapprove all users at once.")
+    else:
+        buttons = InlineKeyboardMarkup([[
+            InlineKeyboardButton(
+                text="Unapprove all users", callback_data="unapproveall_user")
+        ], [InlineKeyboardButton(text="Cancel", callback_data="unapproveall_cancel")]])
+        update.effective_message.reply_text(
+            f"Are you sure you would like to unapprove ALL users in {chat.title}? This action cannot be undone.",
+            reply_markup=buttons,
+            parse_mode=ParseMode.MARKDOWN)
+
+
+@run_async
+def unapproveall_btn(update: Update, context: CallbackContext):
+    query = update.callback_query
+    chat = update.effective_chat
+    message = update.effective_message
+    member = chat.get_member(query.from_user.id)
+    if query.data == 'unapproveall_user':
+        if member.status == "creator" or query.from_user.id in DRAGONS:
+            users = []
+            approved_users = sql.list_approved(chat.id)
+            for i in approved_users:
+                users.append(int(i.user_id))
+            for user_id in users:
+                sql.disapprove(chat.id, user_id)
+
+        if member.status == "administrator":
+            query.answer("Only owner of the chat can do this.")
+
+        if member.status == "member":
```

Comment:
I don't think there is need for two if, You can just say "Only owner of the chat can do this." to both admin and member

Reply:
Just tried to do like Rose.

Reply:
I don't think that changes the fact that It's Invalid info. You are telling to a member that "You need to be a admin" and "You need to be owner" to a admin.